src/pathfinding_service/service.py

```diff
@@ -39,6 +40,16 @@
 log = structlog.get_logger(__name__)
 
 
+class DeferMessage(Exception):
+    """Stop processing the message and handle it when the channel has been opened"""
+
+    deferred_message: DeferableMessage
+
+    def __init__(self, deferred_message: DeferableMessage):
```

Comment:
This should have a `-> None` annotation, strange that mypy doesn't complain.

Reply:
Mypy does not require a `-> None` annotation for `__init__` unless it has not arguments. In that case an annotation is required to tell mypy whether this method is typed or not. If you like I can add it, but I don't see the benefit, personally.

Reply:
Ah right. Fine for me! 👍